Automattic/jetpack
# Contact Form: improve form error message 2

This is a new attempt at https://github.com/Automattic/jetpack/pull/34520 after we've added scripts to the build process. This should ensure strings are properly localized.

## Proposed changes:

This PR improves the error message displayed upon submitting a Contact Form by mentioning the number of errors and the labels of the fields to fix.

In this PR, I'm also refactoring how localized texts are handled. They're now directly defined in the script.

### Other information:


## Jetpack product discussion


Project thread: pf5801-87-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with this branch
- Build the package: `cd projects/packages/forms && npm run build`
- Create a new post
- Add the _Contact Form_ block
- Add several fields, some of which should be required
- Open the preview
- Submit an empty form
- Notice the error message mentions the number of errors and the fields to fix

<img width="500" alt="Screenshot 2023-12-08 at 9 53 04 AM" src="https://github.com/Automattic/jetpack/assets/1620183/225fd201-5771-4165-81d1-e24244dc310c">
